nixos/modules/services/networking/searx.nix

```diff
@@ -10,17 +10,17 @@ let
     cd ${runDir}
 
     # write NixOS settings as JSON
-    cat <<'EOF' > settings.yml
-      ${builtins.toJSON cfg.settings}
-    EOF
+    (
+      umask 077
+      cat <<'  EOF' > settings.yml
+        ${builtins.toJSON cfg.settings}
```

Comment:
Any reason we aren't using an intermediary file here as well?

Wouldn't the string `EOF` in the JSON document break this code?

Reply:
Yeah, it would. I don't know any better way, though: writeText creates an additional file in the nix store that I would have to copy over to /var/ and change its permissions to rw.

Reply:
Yeah but you could do `umask 077 && cp --no-preserve=all ${...} settings.yml` and that wouldn't require changing it to executable.

Reply:
Good idea, thank you!